### ISSUE TITLE
ECONNRESET

### ISSUE THREAD

**Issue by michaelwhill** [iRayanKhan/homebridge-chamberlain#45]
I am able to control two garage doors with no issues. On occasion after the doorstate has changed from closing to closed I receive a FetchError ECONNRESET message for both doors.  Running Node v10.15.0 and Npm 6.5.0 on a RaspberryPI Raspbian 9.6 (stretch).                                             

[1/12/2019, 2:46:36 PM] [Single Car Garage] doorstate changed from opening to open
[1/12/2019, 3:39:00 PM] [Two Car Garage] desireddoorstate changed from closed to open
[1/12/2019, 3:39:02 PM] [Two Car Garage] doorstate changed from closed to opening
[1/12/2019, 3:39:18 PM] [Two Car Garage] doorstate changed from opening to open
[1/12/2019, 3:50:15 PM] [Two Car Garage] doorstate changed from open to closing
[1/12/2019, 3:50:15 PM] [Two Car Garage] desireddoorstate changed from open to closed
[1/12/2019, 3:50:25 PM] [Two Car Garage] doorstate changed from closing to closed
[1/12/2019, 3:58:13 PM] [Single Car Garage] { FetchError: request to https://myqexternal.myqdevice.com/api/v4/DeviceAttribute/GetDeviceAttribute?AttributeName=doorstate&MyQDeviceId=xxxxxxx failed, reason: read ECONNRESET
    at ClientRequest.<anonymous> (/usr/local/lib/node_modules/homebridge-chamberlain/node_modules/node-fetch/index.js:133:11)
    at ClientRequest.emit (events.js:182:13)
    at TLSSocket.socketErrorListener (_http_client.js:392:9)
    at TLSSocket.emit (events.js:182:13)
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  name: 'FetchError',
  message:
   'request to https://myqexternal.myqdevice.com/api/v4/DeviceAttribute/GetDeviceAttribute?AttributeName=doorstate&MyQDeviceId=xxxxxxx failed, reason: read ECONNRESET',
  type: 'system',
  errno: 'ECONNRESET',
  code: 'ECONNRESET' }
[1/12/2019, 3:59:58 PM] [Two Car Garage] { FetchError: request to https://myqexternal.myqdevice.com/api/v4/DeviceAttribute/GetDeviceAttribute?AttributeName=doorstate&MyQDeviceId=xxxxxxx failed, reason: read ECONNRESET
    at ClientRequest.<anonymous> (/usr/local/lib/node_modules/homebridge-chamberlain/node_modules/node-fetch/index.js:133:11)
    at ClientRequest.emit (events.js:182:13)
    at TLSSocket.socketErrorListener (_http_client.js:392:9)
    at TLSSocket.emit (events.js:182:13)
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  name: 'FetchError',
  message:
   'request to https://myqexternal.myqdevice.com/api/v4/DeviceAttribute/GetDeviceAttribute?AttributeName=doorstate&MyQDeviceId=xxxxxxxxx failed, reason: read ECONNRESET',
  type: 'system',
  errno: 'ECONNRESET',
  code: 'ECONNRESET' }

**Comment by janthony6** [comment]
update your api.js line to:

'User-Agent': 'Chamberlain/3.120 (iPhone; iOS 12.2.0; Scale/2.00)', 

current version is using ios 10.0.1 which is outdated so it will throw an error.

**Comment by perarduaadalta** [comment]
12.1.4 is the current version of iOS (not 12.2.0); updated to 12.1.4, still throwing the ECONNRESET error intermittently

**Comment by stackunderflow** [comment]
My logs are full of:
FetchError: request to https://myqexternal.myqdevice.com/api/v4/User/Validate failed, reason: read ECONNRESET
    at ClientRequest.<anonymous> (/homebridge/node_modules/node-fetch/index.js:133:11)

I know Chamberlain did some maintenance, and it looks to me like they're now using some microsoft cloud services.  Some URLs have changed and possibly more.

Homebridge isn't working with MyQ now because of these changes. I tried updating the new User-Agent but that didn't work.  I my have to hold my nose and pay >$100 for a stupid MyQ hardware bridge.

**Comment by iGino** [comment]
Any updates on this? I have the same error constantly appearing in my logs (see below), however, updating the User-Agent in api.js did not resolve the issue.

```
[2019-4-14 00:27:52] [Garage 2] { FetchError: request to https://myqexternal.myqdevice.com/api/v4/DeviceAttribute/GetDeviceAttribute?AttributeName=doorstate&MyQDeviceId=XXXXXXXXXX failed, reason: read ECONNRESET
    at ClientRequest.<anonymous> (/usr/local/lib/node_modules/homebridge-chamberlain/node_modules/node-fetch/index.js:133:11)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at TLSSocket.socketErrorListener (_http_client.js:387:9)
    at emitOne (events.js:116:13)
    at TLSSocket.emit (events.js:211:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
  name: 'FetchError',
  message: 'request to https://myqexternal.myqdevice.com/api/v4/DeviceAttribute/GetDeviceAttribute?AttributeName=doorstate&MyQDeviceId=XXXXXXXXXX failed, reason: read ECONNRESET',
  type: 'system',
  errno: 'ECONNRESET',
  code: 'ECONNRESET' }
```

**Comment by iGino** [comment]
This might be unrelated, but I'm seeing two new errors: ETIMEDOUT & ENETUNREACH

```
[2019-4-14 03:35:13] [Garage 1] { FetchError: request to https://myqexternal.myqdevice.com/api/v4/DeviceAttribute/GetDeviceAttribute?AttributeName=doorstate&MyQDeviceId=XXXXXXXXXX failed, reason: connect ETIMEDOUT 40.78.89.9:443
    at ClientRequest.<anonymous> (/usr/local/lib/node_modules/homebridge-chamberlain/node_modules/node-fetch/index.js:133:11)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at TLSSocket.socketErrorListener (_http_client.js:387:9)
    at emitOne (events.js:116:13)
    at TLSSocket.emit (events.js:211:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
  name: 'FetchError',
  message: 'request to https://myqexternal.myqdevice.com/api/v4/DeviceAttribute/GetDeviceAttribute?AttributeName=doorstate&MyQDeviceId=XXXXXXXXXX failed, reason: connect ETIMEDOUT 40.78.89.9:443',
  type: 'system',
  errno: 'ETIMEDOUT',
  code: 'ETIMEDOUT' }
[2019-4-14 03:36:38] [Garage 1] { FetchError: request to https://myqexternal.myqdevice.com/api/v4/DeviceAttribute/GetDeviceAttribute?AttributeName=doorstate&MyQDeviceId=XXXXXXXXXX failed, reason: connect ENETUNREACH 40.78.89.9:443
    at ClientRequest.<anonymous> (/usr/local/lib/node_modules/homebridge-chamberlain/node_modules/node-fetch/index.js:133:11)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at TLSSocket.socketErrorListener (_http_client.js:387:9)
    at emitOne (events.js:116:13)
    at TLSSocket.emit (events.js:211:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
  name: 'FetchError',
  message: 'request to https://myqexternal.myqdevice.com/api/v4/DeviceAttribute/GetDeviceAttribute?AttributeName=doorstate&MyQDeviceId=XXXXXXXXXX failed, reason: connect ENETUNREACH 40.78.89.9:443',
  type: 'system',
  errno: 'ENETUNREACH',
  code: 'ENETUNREACH' }
```

**Comment by ghost** [comment]
> update your api.js line to:
> 
> 'User-Agent': 'Chamberlain/3.120 (iPhone; iOS 12.2.0; Scale/2.00)',
> 
> current version is using ios 10.0.1 which is outdated so it will throw an error.

This worked perfectly.

**Comment by schwie** [comment]
I updated mine to:

      'User-Agent': 'Chamberlain/3.120 (iPhone; iOS 12.3.1; Scale/2.00)',

and now its working wonderfully.

For folks trying this on macOS, open Terminal and enter the following command, changing [username] to the name of your user folder without the brackets:

pico /Users/[username]/.npm-packages/lib/node_modules/homebridge-chamberlain/src/api.js

In pico (the text editor), change line 17 to:

      'User-Agent': 'Chamberlain/3.120 (iPhone; iOS 12.3.1; Scale/2.00)',

To save changes and exit out:

1. press and hold "control" and "x" keys and then release

2. press "y" key (to save your changes to api.js)

**Comment by caseywebdev** [comment]
Fixed in `1.0.4`
